### ISSUE TITLE
camera_info not published (self is missing when getting a member value)

### ISSUE THREAD

**Issue by mmllee** [ros-drivers/axis_camera#30]
bug in axis.py 
line 128:
cimsg.header.stamp = msg.header.stamp // (exception is thrown here!)
should read
cimsg.header.stamp = self.msg.header.stamp

**Comment by alexcordella** [comment]
Hi all,
I solved this issue.
You have put these lines into the axis.py:
- in "publishFramesContinuously(self):" put
  ....
     self.publishMsg()
     self.publishCameraInfoMsg()
.....
-in "publishCameraInfoMsg(self):" put
....
   cimsg = self.axis.cinfo.getCameraInfo() #SUB
        #cimsg.header.stamp = msg.header.stamp #SUB
        cimsg.header.stamp =self. msg.header.stamp #ADD
        cimsg.header.frame_id = self.axis.frame_id
        cimsg.width = self.axis.width
        cimsg.height = self.axis.height
        self.axis.caminfo_pub.publish(cimsg)  #ADD
.....
Now the camera_info topic is rightly published

Regard,
Alessandro

**Comment by awesomebytes** [comment]
The issue is not very related to its tittle anymore. It's an interesting discussion tho. If anyone makes a PR to fix the camera_info stuff I would be glad to give it a try and merge.

**Comment by jack-oquin** [comment]
@alexcordella: can you render your previous message in some more-readable form?

**Comment by mcorah** [comment]
I submitted a pull request to fix this issue. #38 implements what was originally being discussed and what @alexcordella was describing.

**Comment by alexcordella** [comment]
Open axis.py:
1- Function "publishFramesContinuously(self):" put .... 
self.publishMsg() 
self.publishCameraInfoMsg() .....
2- Function "publishCameraInfoMsg(self):" put .... 
cimsg = self.axis.cinfo.getCameraInfo() 
#cimsg.header.stamp = msg.header.stamp 
 cimsg.header.stamp =self.msg.header.stamp
 cimsg.header.frame_id = self.axis.frame_id
 cimsg.width = self.axis.width 
cimsg.height = self.axis.height
 self.axis.caminfo_pub.publish(cimsg) .... 

Now the camera_info topic is rightly published. Let me know if now is readable.

**Comment by ibaranov-cp** [comment]
I agree with this fix, just worked for me. Makes callibration much more feasible to use now.

**Comment by filiperosa** [comment]
Also worked for me
